src/main/java/io/testproject/sdk/internal/rest/AgentClient.java

```diff
@@ -230,14 +240,35 @@ private AgentClient(final URL remoteAddress, final String token, final Capabilit
 
         // Start reports queue
         if (!disableReports) {
-            this.reportsQueue = new ReportsQueue(this.httpClient);
+            this.reportsQueue = new ReportsQueue(this.httpClient, this.getSession().getSessionId());
             this.reportsQueueFuture = reportsExecutorService.submit(this.reportsQueue);
         }
 
         // Make sure to exit gracefully
         Runtime.getRuntime().addShutdownHook(new Thread(this::close));
     }
 
+    /**
+     * Determine whether the Agent support session reuse.
+     * @return True if session can be reused, otherwise False.
+     */
+    private static boolean sessionCanBeReused() {
```

Comment:
```suggestion
    private static boolean canReuseSession() {
```

Then it'll look better in an `if` statement ;)
```java
if (canReuseSession()) {
    ... reuse the session ...
}
```

Reply:
Changed.